src/modules/Html/ProgressBar.ts

```diff
@@ -39,13 +39,16 @@ export class LoadingBar{
         let _left = this.currentIndex - 1;
         _left = _left < 0 ? 0 : _left;
         let _right = this.bar_length - _left - 1;
-        process.stdout.write(`${this.message}${this.chunksDownloaded} :` + chalk.green(` ${this.empty.repeat(_left)}${this.fill}${this.empty.repeat(_right)}`));
+        if (process.stdout.write)
+            process.stdout.write(`${this.message}${this.chunksDownloaded} :` + chalk.green(` ${this.empty.repeat(_left)}${this.fill}${this.empty.repeat(_right)}`));
```

Comment:
instead of having a bunch of if statements you could have used save navigation.
```suggestion
        process.stdout?.write(`${this.message}${this.chunksDownloaded} :` + chalk.green(` ${this.empty.repeat(_left)}${this.fill}${this.empty.repeat(_right)}`));
```